src/App.tsx

```diff
@@ -5,28 +5,51 @@ import * as RLP from "rlp";
 type AppState = {
   mode: "encoded" | "decoded";
   text: string;
-  rlp: any;
+  rlp: {
+    valid: boolean;
+    decoded: any;
+  };
 };
 
 class App extends React.Component<{}, AppState> {
   public constructor() {
     super({});
+
+    const stock = "0xc701c502c303c104";
     this.state = {
       mode: "encoded",
-      text: "0xc701c502c303c104",
-      rlp: [],
+      text: stock,
+      rlp: {
+        valid: true,
+        decoded: RLP.decode(stock),
+      },
     };
   }
 
   private handleTextChange = (e: any) => {
-    this.setState({ mode: "encoded", text: e.target.value });
+    const { value }: { value: string } = e.target;
+    let rlp;
+    try {
+      rlp = {
+        valid: true,
+        decoded: RLP.decode(value.trim()),
```

Comment:
How about adding "0x" prefix automatically?
It seems there is no way to enter the value except a hex string.

Reply:
I'll create an issue for it.
Let's discuss it on there.